libutils/array_map.c

```diff
@@ -86,6 +86,26 @@ bool ArrayMapRemove(ArrayMap *map, const void *key)
     return false;
 }
 
+bool ArrayMapRemoveSoft(ArrayMap *map, const void *key)
+{
+    assert(map != NULL);
+
+    for (int i = 0; i < map->size; ++i)
+    {
+        if (map->equal_fn(map->values[i].key, key))
+        {
+            map->destroy_key_fn(map->values[i].key);
+
+            memmove(map->values + i, map->values + i + 1,
+                    sizeof(MapKeyValue) * (map->size - i - 1));
```

Comment:
It's just a copy-pasted code from `ArrayMapRemove()`.

Reply:
Actually, we don't have any unit tests doing `ArrayMapRemove()`. I created [CFE-3643[(https://tracker.mender.io/browse/CFE-3643) instead of trying to save the world in this PR.